app/controllers/screenings_controller.rb

```diff
@@ -0,0 +1,58 @@
+class ScreeningsController < ApplicationController
+  before_action :set_screening, only: %i[show edit update destroy]
+  before_action :set_hall, :set_movie, only: %i[new create edit update]
+
+  def index
+    @screenings = Screening.all
+  end
+
+  def new
+    @screening = Screening.new
+  end
+
+  def create
+    @screening = Screening.new(screening_params)
+
+    if @screening.save
+      redirect_to screenings_path
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def show; end
+
+  def edit; end
+
+  def update
+    if @screening.update(screening_params)
+      redirect_to @screening
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @screening.destroy
+
+    redirect_to screenings_path, status: :see_other
+  end
+
+  private
+
+  def set_screening
+    @screening = Screening.find(params[:id])
```

Comment:
in some of the views you are refering to screenings movie and hall, these will be fetched in separate queries impacting performance
you can fetch everything you need in one query by including those associated object like so:
```suggestion
    @screening = Screening.includes(:movie, :hall).find(params[:id])
```
This is not a major issue with one screening but it's good to keep in mind